.github/workflows/main.yml

```diff
@@ -1,6 +1,17 @@
 name: main
 
-on: [push, pull_request]
+on:
+  workflow_call:
+    inputs:
+      build-artifact-name:
```

Comment:
Is this the only way a reusable workflow can access files created from other jobs? I don't like this very much but if there is no other way then I am okay with this.

Reply:
It's the only easy way I think. It needs to be in a separate job to easily run everything in parallel, and the way to pass between jobs is artifacts. I guess it could do it with caching instead, but the same thing would be required, and personally I prefer artifacts because you can look at them directly in case of a test fail.

Reply:
> It's the only easy way I think. It needs to be in a separate job to easily run everything in parallel, and the way to pass between jobs is artifacts. I guess it could do it with caching instead, but the same thing would be required, and personally I prefer artifacts because you can look at them directly in case of a test fail.

What if we use composite action instead of reusable workflow? It looks like we can share workspaces with it.